src/plugins/saved_objects_management/public/triggers/index.ts

```diff
@@ -0,0 +1,9 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+export {
+  SAVED_OBJECT_DELETE_TRIGGER,
+  savedObjectDeleteTrigger,
+} from './saved_object_delete_trigger';
```

Comment:
There currently is no UX flow for this - this is done behind the scenes. It was discussed and decided it will be simpler to simply clean them up in the background when visualization saved objs are deleted.

Reply:
Would it be helpful to add a bit more context on the delete trigger and the reason we currently have it, to clean up augment-vis saved objs when deleting a visualization saved obj?

Reply:
Oh - that's surprising. UI actions is a good event bus, but my mental model is that it's an event bus specifically for user actions. And the saved_object_management plugin is primarily a UI layer on top of saved objects. So if it's all automatic and behind the scenes based on the relationship model, that feels like something that should entirely be handled by the saved_object plugin itself.

Not a blocker, but I'm not sure this ended up with the right separation of concerns.

Reply:
That's fair enough, makes sense to me. This is the first situation of trying to automatically clean up related saved objects that become useless when another saved object is deleted, so there's probably some ways this could be expanded upon and have a clear UI for communicating that. Our idea of deleting automatically was just an optimization to try to help keep an overwhelming amount of stale `augment-vis` saved objects if a user deletes related visualizations.

I will add some of these thoughts in the refactoring issue discussing potentially moving the action to the saved_objects plugin.

Reply:
Discussed offline - will keep as-is. Followup issue to have this expanded to all saved obj types here: https://github.com/opensearch-project/OpenSearch-Dashboards/issues/4499